### PR TITLE
Convert many files to intra-doc links

### PR DESCRIPTION
**Pull request by jyn514**
Helps with https://github.com/rust-lang/rust/issues/75080
r? @poliorcetics 
I recommend reviewing one commit at a time, but the diff is small enough you can do it all at once if you like :)

### CODE REVIEW

**Comment by jyn514** [comment]
https://github.com/rust-lang/rust/pull/76235/commits/cafab51b291405772e55849fd8ab1868746a4c55 is now possible after https://github.com/rust-lang/rust/pull/75652.
https://github.com/rust-lang/rust/pull/76235/commits/2b01df4c2ca56734f16270884b3211bc2e87c82f I think was just missed by your script because of the hash fragments, I converted it with `sed 's@../io/index.html#platform-specific-behavior@crate::io#platform-specific-behavior@' library/std/src/fs.rs  -i`.

I found both of these with `rg '\[.*\]: \.\./.*' library/ | rg -v '\.\./(book|reference|primitive.slice.html)'`.

**Comment by jyn514** [comment]
Actually there are some more links I want to convert, so marking as WIP.

**Comment by jyn514** [comment]
Found most of these using the regex in https://github.com/rust-lang/rust/issues/75080#issuecomment-685285227. Once this and the claimed files are done I think the tracking issue will be almost ready to close :) except for primitives, upstream links, and `core/str/mod.rs`.

**Comment by jyn514** [comment]
Yeah bors is confused and doesn't know what to do about reviewers that aren't part of the rust-lang org. If you like I can assign myself, but that seems a little silly.

**Comment by jyn514** [comment]
Hmm ... the CI failure looks like the same issues as in https://github.com/rust-lang/rust/issues/76106. I'll try to debug that after #75756 lands, I'll revert the changes to `MaybeUninit` in the meantime.

File: library/std/src/fs.rs
Comment (LeSeulArtichaut) [inline library/std/src/fs.rs:556]: Looks like `crate::io` is already imported

```suggestion
    /// [changes]: crate::io#platform-specific-behavior
```

File: library/std/src/sys/unix/ext/fs.rs
Comment (LeSeulArtichaut) [inline library/std/src/sys/unix/ext/fs.rs:11]: This is a intra-doc link bug?

**Review by LeSeulArtichaut — APPROVED** [review]
Looks good to me, left a nit and a question

File: library/std/src/sys/unix/ext/fs.rs
Comment (jyn514) [reply library/std/src/sys/unix/ext/fs.rs:11]: No, associated items from traits are only resolved if the trait is in scope. It's the same rules as normal path resolution, although I don't think it's documented anywhere but https://github.com/rust-lang/rfcs/blob/master/text/1946-intra-rustdoc-links.md#traits-in-scope. If you're interested in documenting it in the rustdoc book that would be very helpful :)

**Comment by LeSeulArtichaut** [comment]
Also I guess you may want to squash your commits before merging?

**Comment by jyn514** [comment]
> Also I guess you may want to squash your commits before merging?

I was planning to keep them as-is, one per file, but I can squash if you prefer.

**Comment by LeSeulArtichaut** [comment]
I don’t have any preference and I don’t even have merge permissions here, I was only *suggesting* this because I know some reviewers like to have everything clean and squashed

File: library/std/src/prelude/mod.rs
Comment (poliorcetics) [inline library/std/src/prelude/mod.rs:78]: I think you changed the link here

File: library/std/src/sys/unix/ext/fs.rs
Comment (poliorcetics) [reply library/std/src/sys/unix/ext/fs.rs:11]: When importing them as `Read as _`, does it work ? It would avoid the `#[allow(unused_imports)]` I thinK.

**Comment by poliorcetics** [comment]
One commit per file does not bother me but this is a doc so squashing won't lose much history or make bisecting harder.

File: library/std/src/sys/unix/ext/fs.rs
Comment (jyn514) [reply library/std/src/sys/unix/ext/fs.rs:11]: The links resolve but it doesn't allow removing `unused_imports`. Let me know if you still want me to switch.

File: library/std/src/prelude/mod.rs
Comment (jyn514) [reply library/std/src/prelude/mod.rs]: Good catch, fixed.

File: library/std/src/sys/unix/ext/fs.rs
Comment (poliorcetics) [reply library/std/src/sys/unix/ext/fs.rs:11]: If it doesn't remove the attribute don't bother, this is clear and less cluttery this way

**Comment by jyn514** [comment]
@poliorcetics did I miss a review comment, is this waiting on anything?

**Pull request merged**
